File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        table={}
        res=[]
        
        for i in range(len(numbers)):
            
            temp = target-numbers[i]
            
            if temp not in table:
                table[numbers[i]]=i
            
            else:
                res.append(table[temp]+1)
                res.append(i+1)
                     
        return res

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        int low=0, high=numbers.size()-1;
        
        vector<int> res(2);
        
        while(low<high){
            
            if(target==numbers[low]+numbers[high]){
                res[0]=low+1;
                res[1]=high+1;
                break;
            }
            else if(target<numbers[low]+numbers[high]){
                high--;
            }
            else{
                low++;
            }
            
        }
        
        return res;
        
    }
};